lib/transport_ssl.c

```diff
@@ -192,8 +192,14 @@ static int ssl_write(transport_handle_t t, const char *buffer, int len, int time
 
 static int ssl_read(transport_handle_t t, char *buffer, int len, int timeout_ms)
 {
-    int ret;
+    int poll = -1, ret;
     transport_ssl_t *ssl = transport_get_context_data(t);
+
+    if (mbedtls_ssl_get_bytes_avail(&ssl->ctx) <= 0) {
+        if ((poll = transport_poll_read(t, timeout_ms)) <= 0) {
```

Comment:
@panitaxx I think this will re-call ssl_read again, I've tried using `select` with timeout for mbedtls fd, but it is not ok, so It was removed. Why you need to call ssl_read again?

Reply:
poll_read does not call ssl_read. it just calls select and if it returns = 0 is just timeout which also is 0 bytes. <0 error. and >0 is the  fds which really does not matters because we only have one fd so it gets to the next line to read because we now know we have something to read.

Reply:
Ah, thank you @panitaxx, This will resolve the problem when calling `transport_poll_read` before reading.